### DISCUSSION TITLE
Love this project. Hope it's available soon.

### DISCUSSION THREAD

**Discussion by Pagandy** [peterantypas/maiana#44] [General]
Hi Peter. Great to see someone working on this. Feels like an obvious project for opensource development given the commercial options. And I'm really impressed with the design decisions you've taken. I'm interested as soon as a kit is available. How can I get notified when you have units ready to ship? Do you have a mailing list? 

Thanks.

**Comment by trimaranrob** [comment]
I am very interested in this for my sailboat as well!
If there is a mailing list please add me.

**Comment by peterantypas** [comment]
I haven't organized the business side of this yet, but I will have a mailing list at some point. My daytime job is now demanding a whole lot more of my time than in the past few months, so that will chip away at this project too.

**Comment by EdCafferata** [comment]
In also have intres in it, Please let me know to.
Want to help aishub with some receiver 😀

**Comment by javelin911** [comment]
Agreed, Great to see this work!  Sign me up for a system!!!

**Comment by tortfeaser** [comment]
Yep, great project. If there's a list, please put me on it.

**Comment by revolteur** [comment]
Hi @peterantypas Awesome Project! Thank you for publishing it to the community. Really appreciated the design decisions you made, like keeping all RF outside! Nice. Curious to know what considerations kept you from make it an all in one outdoor system. (Input: 12V | Output: NMEA Data over wifi).

Btw - I come across couple of rendering you published for the RJ45 breakout board that supports a pin out for an ESP32 to broadcast sensor data over wifi, however the published CAD files 10.0.01 are not reflecting this update. Did you drop this idea? I am planing on ordering couple of PCB's in a hopeful attempt to test them out on my boat.

Btw - when uploading the transponder-10.0.01.brd file to EasyEDA it errors out. Not the case for the Rj45 breakout board.
I managed to import the project to kicad as a non-kicad project. Few layers has some errors. So far so good. Sticking to open source for now.

Sign me up please to you list in case my attempts to assemble one fails :)
Haven't done this for a long while. Thanks again for this great project!

**Reply by peterantypas** [reply]
I'm on the fence about making MAIANA&trade; fully wireless. My main concern is that it adds another layer of complexity to the existing board and a potential point of failure. 

But like I said, I'm open to the idea, as long as whatever I add is still open source and has a compatible license. The WiFi module/SoC would need to be a standalone UART <-> TCP/IP adapter. The existing MCU doesn't have much processor bandwidth. Do you have any particular ones in mind?

**Reply by peterantypas** [reply]
This looks like a simple enough base project for ESP32, and the use case is NMEA: https://github.com/AlphaLima/ESP32-Serial-Bridge/blob/master/ESP32-Serial-Bridge.ino

I'll look into designing a 2.4GHz antenna on the opposite end of the PCB. It may not be ideal given the proximity to the cable, but there's no other place for it. It may need to be a chip antenna instead of a meandering "F". I'm positive I don't have enough room for those.

**Reply by peterantypas** [reply]
... if the UART and other lines do go away, then so do all the TVS diodes and buffers near the MCU. That frees up quite a bit of room. Also, since the cable will only need 2 conductors, it can be something much thinner than Cat-5. With the right strain relief it could be positioned at a consistent enough distance from the antenna to maintain some reasonable tuning. I'm starting to like this idea ...

**Reply by peterantypas** [reply]
I really like Adafruit's design with the Fractus antenna: https://www.adafruit.com/product/4335#technical-details

That's a good baseline for the hardware design.

**Comment by marcelrv** [comment]
https://github.com/peterantypas/maiana/issues/13#issuecomment-683385528 

This is a well working serial-TCP  esp based firmware
>  what I personally use on my boat is from this project https://github.com/jeelabs/esp-link
> It normally connects to the accespoint (openplotter project). However if it can't find it it becomes a hotspot itself.
> It has basic webpage to allow the configuration, e.g Search for accespoints and setting password etc.

**Reply by peterantypas** [reply]
OK, I'll look into this as well.

**Comment by peterantypas** [comment]
One of my concerns with going wireless is the lack of simple button controls. How would a user disable transmission, for instance? 

I think this will require a custom web UI for MAIANA&trade;, so that one can:

a) provision MMSI, name, call sign, etc via a mobile phone
b) enable / disable transmission via their phone

Therefore, I am not enthused with the jeelabs project. This needs to be a custom firmware, built from scratch, with a branded web UI and laser focused on the problem. The jeelabs project is trying to be everything to everyone and it's getting bloated.

Another challenge with eliminating the RJ45 is the assembly method that I devised is no longer viable. With the RJ45, it is very easy to insert the cable and snap it into the PCB jack from the bottom of the tube, after full assembly and sealing of the unit. It's also easy to insert a finger, push the tab and release the cable. I can't think of a 2-wire connector that offers this simple feature. Maybe there's a Molex or JST part that can do this?

This assembly approach allows the unit to be easily removed by cutting one heatshrink tube (the one sealing the bottom), then lifting it and removing the RJ45 cable (as long as there's enough slack). If the cable cannot be easily detached and reattached, it will have to be removed from the cabin side, then pulled through the railing or antenna mast and re-installed after servicing. That's a pain in the ass. I don't want the user to ever have to repeat this process. This is why I chose the RJ45 in the first place.

**Reply by peterantypas** [reply]
A tertiary concern is heat dissipation. I have never used an ESP32, but given its current consumption, I suspect it puts out quite a bit of heat. Can someone here offer some insight?

**Reply by peterantypas** [reply]
Also, if the design does end up going wireless, I can imagine the next step being cutting the cord and going battery powered. Having the current draw of an ESP32 will make it hard to get a decent battery life, no matter what cells I put in there.

I have spent the last year working on an extended range BLE -> LTE gateway for a medical application, and I really like the idea of a BLE peripheral as a NMEA source via a NOTIFY characteristic. It makes a lot more sense than WiFi in my mind, and the power consumption of a typical BLE radio acting as a peripheral is nothing compared to an ESP32. That said, I haven't seen any standard for NMEA over BLE. I would be very interested in exploring that.

**Reply by marcelrv** [reply]
> A tertiary concern is heat dissipation. I have never used an ESP32, but given its current consumption, I suspect it puts out quite a bit of heat. Can someone here offer some insight?

I just pointed my IR thermometer at ESP, running a BLE-wifi gateway. It is consuming 15mA, running at 27C (80F, about 2C more than the current room temp). The highest temp on the particular PCB is the memory chip, which was running 32C

For battery driven applications, it may consume too much. Don't know about the consumption if it is soly in BT mode.
Having said that, don't know if similar wifi solutions consume much less...
The sweet thing with the ESP's is that they support both wifi and BLE hence it gives a lot of options for connectivity.

A more dedicated environment / custom web UI for MAIANA™ is obvious even better, but indeed requires a whole different level of effort.

**Reply by revolteur** [reply]
There is also this Autoconnect library: https://hieromon.github.io/AutoConnect/index.html
Yes, the jeelabs is more like a Swiss knife library, the purpose build route is always interesting.

**Reply by revolteur** [reply]
I haven't came across any BLE NMEA project either. The BLE power consumption is certainly more attractive. 

In the context of this project wouldn't BLE create a limitation though. For example, how would a user running OpenCPN or iNavX on a ipad receive data from MAIANA™  and other sensors over BLE.  iNavX for example support only TCP/IP data over wifi.  In a closed system, (2 unit design) BLE would be better then Wi-Fi.  The 1 unit design not so much I think.

**Comment by revolteur** [comment]
The wireless, No-Cat  route, certainly adds an new layer to the stack and project. I imagine it introducing as well a new set of requirements (HW, FW, UX, Config, etc)  to address common boat setup use cases, like boats with an existing wifi gateway vs. those with none. This is needed to ensure the project remain accessible to all users. 

The obvious benefit with the wireless route is eliminating the RJ45 cable and the auxiliary board all together. The installation would require only power to the transponder unit since all I/O would be over RF ( Wi-Fi / VHF) 

Tx OFF can be handled via a soft power button in the web UX (pirate mode). I would personally still add kill switch on the power line near the battery or a dedicate switch on the DC panel of the boat if your boat has one. 

Thinking through the UX perspective and during the installation stage, the captain will need to configure the system via a custom splash capture page: 

- AP mode vs Client Mode (boat w/ wifi gateway) with the respective AP name and password depending on the mode of use selected. 

- Data broadcast: N2k, Nema183 or SignalK. 

- MMSI, name, call sign as you shared as well. 

The wireless route will require OTA update to the FW since the USB is no longer available.  From what I gathered so far researching this topic, the Jeelab esp link library features OTA updates and has a web console to debug/interface the MCU.  

I don’t have a specific wifi SoC in mind, the esp8266, ESP32 seem to have become the 
defacto open source options for  UAV/IOT and their reliability seems to improve. Its hard to beat the form factor of the TinyPico Nano as well. 

Picking a Wifi SoC will have to do more with the available FW features of the open source libraries since coding all this can be extensive. 

I plan on looking into the espressif IDF open source libraries as it packs a bunch of features and they also seem to have  open source web/android/iOS apps. More investigation is needed. Ordered two ESP32 and plan on checking it out. I am in crawl mode so will see how far I get. 

Just wanted to share some of my thoughts on going down the No-Cat route. It certainly comes at a cost and has its own trade offs. Overall I think it will make this project a more modern solution if we manage to pull it off. 

Time to go sailing, will post back.

**Reply by peterantypas** [reply]
That's what I did too:

![IMG_1022](https://user-images.githubusercontent.com/1565933/110353379-dc009580-7feb-11eb-925b-489d9ef3017a.jpg)

**Comment by Pagandy** [comment]
Great that you're back to developing this.

I've noticed that IAS transponders on other boats broadcast that they are
underway so presumably there are other states broadcast. Underway motoring,
underway sailing, anchored.
Also ship length.

Looking forward to seeing how this develops.

Best Regards

Andy Pag

--
Need more?
@andybutshorter
BTC: 3JmkFBF6Y26Kp9j49DjQSf5vcncJJfrDmT
--

**Comment by peterantypas** [comment]
Oh, OK then. It’s the 500mA spec in the data sheet that scared me.

I think it’s definitely worth a discussion going that route.   I don’t love the breakout box. It bothers me that I need 2 PCBs plus an intricate custom enclosure.

I think an MVP here would need to do this:

- Boot up as an AP and offer a captive portal to configure the network and transponder particulars 

- Once the station has been configured, the UI offers status information and basic controls like ‘silent mode’. I don’t want to make it easy for people to spoof MMSI and change it constantly. A ‘factory reset’ will be required to reconfigure everything 

- The main mode of operation is to either push NMEA data by connecting to some server, or accept a number of client connections that it pushes data to.

- An alternate mode might be UDP broadcast or IGMP multicast, but only if there’s demand for it

- The network module will be responsible for all FOTA, including MCU firmware update 

Anything else for the MVP?

**Reply by revolteur** [reply]
I think this captures it and is a good starting point as the discussion evolve. 
From a product design perspective, you can say I share the same feeling about the 2 board solution :) 
An all-outdoor, 1 unit design path is more fitting from my perspective at least.

**Reply by marcelrv** [reply]
> - An alternate mode might be UDP broadcast or IGMP multicast, but only if there’s demand for it

This is what some apps are using e.g. navionics listens to UDP on port 2000. I think they refer to it as digiyacht (see comment bottom page)
https://www.zeilwereld.nl/wp-content/uploads/2019/10/Boating_app_AIS_Devices.pdf

> If you have an AIS device not listed above, you can try connecting it to the Boating app provided that it allows setting Wi-Fi configuration manually. Use one of the configurations below:
> 1. IP Address: 0.0.0.0 (or leave it empty), Port: 2000, Protocol: UDP.
> 2. IP Address: 192.168.15.1, Port: 39150, Protocol: TCP

**Comment by peterantypas** [comment]
Here is where an ESP32 PICO MINI module would go:

![Screenshot from 2021-03-07 20-18-45](https://user-images.githubusercontent.com/1565933/110273853-58f72500-7f82-11eb-8bfb-80a16795e4c4.png)

It fits, but the antenna area hanging off the board will need clearance from metal. The power cable will run horizontally along the lower edge of the board toward some yet-to-be determined connector. It's not ideal, but there's no other option.

**Reply by revolteur** [reply]
Nice!

Have you came across this https://www.tinypico.com/tinypico-nano
Should be smaller footprint than the pico and could free up few mm's
Not sure if you had other reasons for selecting the pico footprint.

**Reply by muebau** [reply]
Hello,
I think the WiFi part might be nice but a drawback too 😂

There might be an existing bus on a vessel with automatic steering, VHF (ATIS, DSC), depth sensor. Usually they would be connected with NMEA0183 or NMEA2k.

In the end it might all end up on a OpenCPN or tablet with navionics etc. but this is just a part of it. I would use the GPS part of this project to provide the position data to the VHF with DSC and use the existing (very) old Garmin GPSMAP as a simple display with depth map in the cockpit.

If MAIANA would become completely wireless it could provide its data to my phone with navionics etc. but I would need a WiFi AP to a WiFi to NMEA2k bridge to feed the data back to the bus.

I think there are very good reasons for the simple solution with cable and the box and very good reasons for the WiFi/BLE solution.

just my 0.02 Euro 😉

**Reply by revolteur** [reply]
True, for boats with an existing NMEA bus, the wireless route will require a Wi-Fi 2 NMEA adapter of sorts would be needed to get the AIS/GNSS data on the existing NMEA bus. Given the different boat setups, hard to derive a one size fit all solution.

**Comment by Rudder2** [comment]
I to love your project and have been waiting for your project to be available for purchase. I like the wired setup. My plan is everything to be wired and than have a WiFi access point for tablet in cockpit to use my navigation / smart boat system and non waterproof displays at the nav station and be be able to display the info in the 50 inch TV in the main saloon. Tablets are chapter that a IP65 or 66 display and i can keep a couple as backup.

My entire boat is going to be open source electronics.

**Reply by revolteur** [reply]
I share the same goals on keeping things open source on my boat as well where possible. Curious, what do you plan on using to interface the AIS/GNSS/etc to your WiFi access point to broadcast sensor data to a tablet?  OpenCPN/Pi or else?

**Reply by marcelrv** [reply]
OpenPlotter is a nice project combining OpenCPN with SignalK / Kplex and a bunch of other useful tools with ease of installing. I I can very much recommend it. I contributed to the SignalK project part of the TCP logic, so it works fine for me now.

In my case I wanted to avoid the annoyance of running wires everywhere, hence I have few ESP based NMEA(serial) -to-WIFI converters that connect regular plotter/GPS, depth etc to openplotter (and to each other)

Besides that, I have both nvmaps & navionics running on a tablet which connect to OP as well. (I don't have proper maps showing depth for OpenCPN yet so unfortunately can't go fully opensource yet)

**Reply by muebau** [reply]
I tried to go as OpenSource as possible too (don't mind the existing Garmin GPSMAP).

I tried to replace the NASA Marine Stingray Echo Sounder with some OpenHardware or uC solution. The transducer should be fine but I cant find a simple way to generate a ping and receive the echo.
If somebody has found something for that please give my a hint 😉

**Reply by peterantypas** [reply]
I did build a prototype depth sounder for the ancient transducer on my boat, and it was accurate to 65 feet. That said, it lived mostly on a breadboard and didn't last very long :P

Not something I want to resurrect right now, but the main idea is to push a high-power PWM signal to the transducer via an impedance matching transformer, then listen for the echo via bandpass filter and multiple low-noise opamp gain stages (I think I used AD797s), plus a peak detector and/or the ADC of the MCU.

To determine the impedance of my transducer, I hooked up a scope to capture the voltage output pattern of the current depth sounder, then used an inline resistor to capture the current pattern. Nowadays I would use my NanoVNA 2 :)

**Reply by muebau** [reply]
Do you have anything left from this design? I am afraid to blow up my transducer or uC (or both) as "analog" stuff is more or less out of my comfort zone. 😂 My studies covered very less about this.

**Reply by Rudder2** [reply]
I plan on using a NMEA 2000 Depth, Speed, Temp transducer.  All the data on my boat will feed into OpenPlotter running on a Raspberry PI 4 with BoatIO hat to accept all my NMEA 2000, NMEA 0182, Analog, and Digital signals from my vessel.  With Boat IO I plan on taking the engine readings right from the stock sending units, monitor bilge pump duty cycle, battery voltages, tank levels, and rudder position to have on the Network also.  My main ships GPS I planned on being the MAIANA AIS system but until that is available I will be have to get the Boat IO with GPS and using it, only like $40 more so not too bad, and that will give me a backup GPS on the network once MAIANA is installed.

Here in the USA NOAA charts are a free download.  I have to do more research on how to get international charts for OpenCPN.  If I have to pay for them I have no problem with that.

I understand the use for WiFi but I prefer hard wiring.  I even ran Cat 6 Ethernet threw my house for any device that isn't mobile...LOL  I'm old school when it comes to communications.  Like I said, I will have WiFi to communicate with the Tablets for controlling OpenPlotter from cockpit and the PyPilot computer.  I know OpenPlotter can take the place on the PyPilot PI but I prefer to have a separate system for AutoPilot / weather instruments that isn't hard wired together just in case the navigation / monitoring system goes down I can till use apparent wind and compass navigation.  I purchased the PyPilot and Weather computer directly from Sean, the inventor, to help support his project.  

What I love about this system I can keep a stack of PIs, reprogrammed SD cards, a couple tablets, and a Bluetooth GPS in a Faraday cage on board to be able to get the the navigation/autohelm system back up in the event of lightning strike or other failure.  Also in the even of a lightening strike I'm only out $1K instead of $10K if I go commercial electronics.

Have anymore questions let me know.  I don't know alot because I'm learning as I build this system but I will help where I can.
